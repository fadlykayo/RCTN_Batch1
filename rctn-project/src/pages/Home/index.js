import React, { Component } from 'react';
import { connect } from 'react-redux';
import axios from "axios";
// import PropTypes from 'prop-types';

import {
  UserActionCreator,
  ProductActionCreator
} from '../../actions';
import {
  Profile,
  Section,
  PastMeetup,
} from '../../components/Organisms';
import {
  SectionContent,
} from '../../components/Molecules';
import {
  SectionTitle,
} from '../../components/Atoms';

import './style.css';

const mapStateToProps = (state) => {
  return {
    members: state.user,
    products: state.product
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchData: () => dispatch(UserActionCreator.fetchData()),
    register: (data) => dispatch(UserActionCreator.register(data)),
    login: (data) => dispatch(UserActionCreator.login(data)),
    logout: () => dispatch(UserActionCreator.logout()),
    createProduct: () => dispatch(ProductActionCreator.createProduct()),
  };
};

class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tanggal: new Date().toString(),
      info: {
        location: 'Jakarta, Indonesia',
        date: '29 January 2019',
        numberOfMembers: '1078',
        headOrganizer: 'Hacktiv8',
        memberOrganizer: 'Best, Twinky, Winky, Dipsy, Lala, Poh. JakartaJS Organizer.',
      },
      twitter: '@JakartaJS',
      hashTag: '#jakartajs',
      meetups: [
        {
          id: 39,
          date: '27 November 2017',
          topic: 'JakartaJS with kumparan',
          participants: '139'
        },
        {
          id: 40,
          date: '27 September 2017',
          topic: 'JakartaJS with bibli',
          participants: '239'
        },
        {
          id: 41,
          date: '27 October 2017',
          topic: 'JakartaJS with Hacktiv8',
          participants: '39'
        }
      ],
      members: [],
      user: {
        name: '',
        email: '',
      },
      resultUser: ''
    };
  };

  componentDidMount() {
    // axios
    // .get("https://randomuser.me/api/?results=5")
    // .then(response => {
    //   this.setState({
    //     members: response.data.results,
    //   })
    // });
    //
    // this.props.fetchData();

    // const userData = localStorage.getItem('user');

    // console.log('localStorage userData', userData);

    // if (userData) {
    //   this.setState({ user: JSON.parse(userData) });
    // };
  };

  onClickLogin = (data) => {
    this.props.login({
      name: 'fadly',
      email: 'fadly.kayo@gmail.com',
    });
  };

  onClickLogout = () => {
    this.props.logout();
  };

  onChangeText = (dynamicKey) => (event) => {
    let user = {...this.state.user};

    user[dynamicKey] = event.target.value;
    // [dynamicKey]: event.target.value,

    this.setState({user});
  };

  handlePost() {
    axios
    .post("http://localhost:4000/daftarNama", {
      "user": {
        "name": this.state.user.name,
        "email": this.state.user.email,
      }
    })
    .then(response => {
      this.setState({resultUser: response.data.user})
    });
  };

  render() {
    // console.log('ini state members', this.state.members);
    // console.log('ini state user', this.state.user);
    console.log('ini props reducer members', this.props.members);
    // console.log('ini props reducer members.user', this.props.members.user);
    return (
      <div className="body">

        <Profile
          classNameSection={'section_right'}
          classNameInfo={'row_right_info_content'}
          leftList={['Location', 'Members', 'Organizer']}
          rightList={[this.state.info.location, this.state.info.numberOfMembers, this.state.info.headOrganizer]}
          title={'Hacktiv8 Meetup'}
          button
          onClick={this.onClickLogin}
        />

        <Section>

          <SectionTitle title={'Next Meetup'}/>

          <SectionContent className="section_content_column_grey">
            <h4 style={{marginTop: 0, marginBottom: 10,}}>Awesome meetup and event</h4>
            <div>{this.state.info.date}</div>
              <div>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</div>
              <div>Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.</div>
              <div>It has survived not only five centuries, but also the leap into electronic typesetting.</div>
              <div>It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages.</div>
            <div>Lorem Ipsum is simply dummy text of the printing.</div>
            <p>See you there!</p>
            <p>{this.state.info.memberOrganizer}</p>
          </SectionContent>

        </Section>

        <Section>

          <SectionTitle title={'About Meetup'}/>

          <SectionContent className="section_content_column_white">
            <div style={{marginBottom: 15,}}>Come and meet other developers interested in Javascript and it's library in Greater Jakarta area.</div>
            <div>Twitter {this.state.twitter} and we use hashtag #jakartajs</div>
          </SectionContent>

        </Section>

        <Section>

          <SectionTitle title={'Members'}/>

          <Profile
            classNameSection={'section_right_alt'}
            classNameInfo={'row_right_info_content_alt'}
            leftList={['Fadly Kayo']}
            rightList={['4 others']}
            title={'Organizers'}
            circleThumbnail
          />

        </Section>

        <PastMeetup
          meetups = {this.state.meetups}
        />

      </div>
    );
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
