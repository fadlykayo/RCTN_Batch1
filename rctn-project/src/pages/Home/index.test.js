import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import Home from './';
import store from "../../configs";

test('Home renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home/>
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
