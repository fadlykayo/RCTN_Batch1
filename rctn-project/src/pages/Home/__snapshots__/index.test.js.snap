// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="body"
>
  <div
    className="row"
  >
    <img
      alt="placeholder"
      src="https://via.placeholder.com/200x200"
    />
    <div
      className="section_right"
    >
      <h3
        className="section_title_right"
      >
        Hacktiv8 Meetup
      </h3>
      <div
        className="section_content_right"
      >
        <div
          className="row_right_info_content"
        >
          <div>
            Location
          </div>
          <div>
            Members
          </div>
          <div>
            Organizer
          </div>
        </div>
        <div
          className="row_right_info_content"
        >
          <div>
            Jakarta, Indonesia
          </div>
          <div>
            1078
          </div>
          <div>
            Hacktiv8
          </div>
        </div>
      </div>
      <button
        className="button"
        onClick={[Function]}
      >
        Join Us
      </button>
    </div>
  </div>
  <div
    className={undefined}
  >
    <h3
      className={undefined}
    >
      Next Meetup
    </h3>
    <div
      className="section_content_column_grey"
    >
      <h4
        style={
          Object {
            "marginBottom": 10,
            "marginTop": 0,
          }
        }
      >
        Awesome meetup and event
      </h4>
      <div>
        29 January 2019
      </div>
      <div>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      </div>
      <div>
        Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.
      </div>
      <div>
        It has survived not only five centuries, but also the leap into electronic typesetting.
      </div>
      <div>
        It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages.
      </div>
      <div>
        Lorem Ipsum is simply dummy text of the printing.
      </div>
      <p>
        See you there!
      </p>
      <p>
        Best, Twinky, Winky, Dipsy, Lala, Poh. JakartaJS Organizer.
      </p>
    </div>
  </div>
  <div
    className={undefined}
  >
    <h3
      className={undefined}
    >
      About Meetup
    </h3>
    <div
      className="section_content_column_white"
    >
      <div
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        Come and meet other developers interested in Javascript and it's library in Greater Jakarta area.
      </div>
      <div>
        Twitter 
        @JakartaJS
         and we use hashtag #jakartajs
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <h3
      className={undefined}
    >
      Members
    </h3>
    <div
      className="row"
    >
      <img
        alt="thumbnail_circle"
        src="https://via.placeholder.com/80x80"
        style={
          Object {
            "borderRadius": 50,
            "height": 80,
            "width": 80,
          }
        }
      />
      <div
        className="section_right_alt"
      >
        <h3
          className="section_title_right"
        >
          Organizers
        </h3>
        <div
          className="section_content_right"
        >
          <div
            className="row_right_info_content_alt"
          >
            <div>
              Fadly Kayo
            </div>
          </div>
          <div
            className="row_right_info_content_alt"
          >
            <div>
              4 others
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="PastMeetup"
    >
      <div
        className="Subtitle"
      >
        Past Meetup
      </div>
      <div
        className="SeeAll"
        onClick={[Function]}
      >
        See All
      </div>
      <div
        className="CardGroup"
      >
        <div>
          <div
            className="CardItem"
          >
            <h3
              className="bold"
            >
              27 November 2018
            </h3>
            <hr />
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="#39"
              />
              <div
                className="bold"
                content="JakartaJS with kumparan"
              />
            </div>
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="139"
              />
              <div
                content="went"
              />
            </div>
            <button
              className="view"
            >
              View
            </button>
          </div>
        </div>
        <div>
          <div
            className="CardItem"
          >
            <h3
              className="bold"
            >
              27 November 2018
            </h3>
            <hr />
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="#40"
              />
              <div
                className="bold"
                content="JakartaJS with bibli"
              />
            </div>
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="239"
              />
              <div
                content="went"
              />
            </div>
            <button
              className="view"
            >
              View
            </button>
          </div>
        </div>
        <div>
          <div
            className="CardItem"
          >
            <h3
              className="bold"
            >
              27 November 2018
            </h3>
            <hr />
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="#41"
              />
              <div
                className="bold"
                content="JakartaJS with Hacktiv8"
              />
            </div>
            <div
              className="make-inline"
            >
              <div
                className="space5px bold"
                content="39"
              />
              <div
                content="went"
              />
            </div>
            <button
              className="view"
            >
              View
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
